e2etest/pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"go.universe.tf/metallb/e2etest/pkg/executor"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+// ValidateOnPrometheus checks the existance of the given metric directly on prometheus pods.
+func ValidateOnPrometheus(exec executor.Executor, query string, expected CheckType) error {
+	fmt.Println("Checking ", query, expected)
+	url := fmt.Sprintf("localhost:9090/api/v1/query?%s", (url.Values{"query": []string{query}}).Encode())
+	stdout, err := exec.Exec("wget", "-qO-", url)
+	if err != nil {
+		return err
+	}
+	// check query result, if this is a new error log it, otherwise remain silent
+	var result PrometheusResponse
+	if err := json.Unmarshal([]byte(stdout), &result); err != nil {
+		return fmt.Errorf("unable to parse query response for %s: %v", query, err)
+	}
+	metrics := result.Data.Result
+	if result.Status != "success" {
+		data, _ := json.MarshalIndent(metrics, "", "  ")
+		return fmt.Errorf("promQL query: %s had reported incorrect status:\n%s", query, data)
+	}
+
+	if expected == NotThere && len(metrics) > 0 {
```

Comment:
is it possible to extend the other metrics validation functions to use a similar semantic?
the current negative tests expect an error for the other validation functions but noerror for this one (when you're setting metrics.NotThere), I think it would be nice if they all behave the same

Reply:
Tried to give a look at them, it would basically mean adding a "thisWouldReturnAnError" parameter, not sure it'd make the codebase better or worse.